ui-admin/src/forms/designer/split/SplitFormElementDesigner.tsx

```diff
@@ -0,0 +1,102 @@
+import {
+  FormContent,
+  FormElement,
+  PortalEnvironmentLanguage,
+  Question,
+  surveyJSModelFromFormContent
+} from '@juniper/ui-core'
+import React, { memo, useState } from 'react'
+import { IconButton } from 'components/forms/Button'
+import { faCode } from '@fortawesome/free-solid-svg-icons'
+import { ListElementController } from 'portal/siteContent/designer/components/ListElementController'
+import { Survey as SurveyComponent } from 'survey-react-ui'
+import { isEqual } from 'lodash'
+import { FormElementEditor } from './FormElementEditor'
+import { FormElementJsonEditor } from './FormElementJsonEditor'
+
+/* Note that this component is memoized using React.memo
+ * Since survey pages can contain many elements, we need to be mindful of
+ * how many times we re-render these components. Since the parent component state
+ * is updated with every keystroke, we memoize this to minimize the number
+ * of re-renders that take place. The SurveyComponent from SurveyJS in particular
+ * is sluggish when undergoing many simultaneous re-renders.
+ */
+
+export const SplitFormElementDesigner = memo(({
+  elementIndex, element, currentPageNo, editedContent, onChange, currentLanguage, supportedLanguages
+}: {
+    elementIndex: number, element: FormElement, currentPageNo: number,
+    currentLanguage: PortalEnvironmentLanguage, supportedLanguages: PortalEnvironmentLanguage[],
+    editedContent: FormContent, onChange: (newContent: FormContent) => void
+}) => {
+  const [showJsonEditor, setShowJsonEditor] = useState(false)
+
+  // Chop the survey down to just the specific question that we're editing, so we can display
+  // a preview using the SurveyJS survey component.
+  const surveyFromQuestion = {
+    title: 'Question Preview',
+    pages: [{ elements: [element] }],
+    questionTemplates: editedContent.questionTemplates
+  }
+  const surveyModel = surveyJSModelFromFormContent(surveyFromQuestion)
+
+  surveyModel.showInvisibleElements = true
+  surveyModel.showQuestionNumbers = false
+
+  // @ts-ignore
+  const elementType = element.type === 'panel' ? 'panel' : 'question'
```

Comment:
I wonder how to make it easy in the UX to move questions into and out of panels. For example, I could see someone cleaning up ATs surveys spending a lot of time moving related questions into panels so that they're grouped together since the export exports without any concept of panels. Right now, in the UX, I'm not sure how we would make that functionality visible. Maybe a `Move Question` button that shows all the panels and questions and you can click where it should go?

I could also see them needing to move translations around - some way of doing "copy all translations for this label" and "paste all translations" could be nice - maybe next to every text field is a json button that converts the text field into a larger one with the translation object? that way you can copy/paste it and see all the translations for one question at a time without diving into the whole question's json?

Idk just some thoughts on things I have felt were missing while we're still early on 😄

Reply:
yeah, related is a "copy question" button for all the times where you're changing the question text but giving the exact same multiple choice options